R/onevax_xvw.R

```diff
@@ -46,37 +47,82 @@ vax_params_xvw <- function(vea = 0, vei = 0, ved = 0, ves = 0,
   assert_scalar_positive(t_stop)
 
   # waned vaccinees move to own stratum, and are not eligible for re-vaccination
-  # 1:x -> 2:v <-> 3:w
-  i_eligible <- 1
-  i_v <- 2
-  i_w <- n_vax <- 3
+  # unvaccinated 1:n_diag_rec (x)
+  # vaccinated from n_diag_rec+1 to 2*n_diag_rec (v)
+  # waned from  2*n_diag_rec+1 to 3*n_diag_rec (w)
+
+  # generate indices for all strata and
+  idx <- stratum_index_xvw_trial(1, n_diag_rec)
+  n_vax <- idx$n_vax
+
+  i_v <- idx$V
+  i_w <- idx$V + n_diag_rec
+  i <- seq_len(idx$n_vax)
+
+  # strata people are diagnosed from
+  i_diagnosedfrom <- i[i %% n_diag_rec != 0]
+
+  # strata people are diagnosed to
+  i_diagnosedto <- i[i %% n_diag_rec != 1]
+
   n_group <- 2
 
+  # create diagnosis history mapping
+  diag_rec <- create_vax_map_branching(idx$n_vax, c(1, 1), i_diagnosedfrom,
+                                       i_diagnosedto, set_vbe = FALSE, idx)
+
+  # waned vaccinees move to own stratum, and are not eligible for re-vaccination
+  i_eligible_temp <- c(1:n_diag_rec)
+  i_v_temp <- c((1 * (n_diag_rec) + 1): (2 * (n_diag_rec)))
```

Comment:
```suggestion
  i_v_temp <- i_v[1:n_diag_rec]
```
Is this clearer what's going on?